src/commandsAndMenu.tsx

```diff
@@ -606,6 +606,16 @@ export function addCommands(
             })`;
             modelIsLoading.reject(msg);
           }
+
+          const slotRegistered = gitModel.statusChanged.connect((_, status) => {
+            const targetFile = status.files.find(
+              fileStatus => model.filename === fileStatus.from
+            );
+            if (!targetFile || targetFile.status === 'unmodified') {
+              mainAreaItem.dispose();
+            }
+          });
+          console.log(slotRegistered);
```

Comment:
Perfect :clap: 

I presume the `console.log` comes from your test to understand the code.

```suggestion
          gitModel.statusChanged.connect((_, status) => {
            const targetFile = status.files.find(
              fileStatus => model.filename === fileStatus.from
            );
            if (!targetFile || targetFile.status === 'unmodified') {
              mainAreaItem.dispose();
            }
          });
```